apis/metal3.io/v1alpha1/baremetalhost_types.go

```diff
@@ -278,6 +278,28 @@ type RAIDConfig struct {
 	SoftwareRAIDVolumes []SoftwareRAIDVolume `json:"softwareRAIDVolumes,omitempty"`
 }
 
+// FirmwareConfig contains the configuration that you want to configure BIOS settings in Bare metal server
+type FirmwareConfig struct {
+	// Whether reset firmware settings.
+	// NOTE: IRMC does not support reset firmware settings.
+	ResetSettings bool `json:"resetSettings,omitempty"`
+
+	// Supports the virtualization of platform hardware.
+	// This supports following options: true, false.
+	// +kubebuilder:validation:Enum="true";"false"
+	VirtualizationEnabled string `json:"virtualizationEnabled,omitempty"`
```

Comment:
The option we agreed on [in the design doc](https://github.com/metal3-io/metal3-docs/blob/master/design/baremetal-operator/bios-config.md) was a boolean to *disable* virtualization.

I get that using strings allows us to distinguish between the case where we want to overwrite the BIOS settings with a particular value (either true or false) and the one where we want to leave whatever settings have been manually configured by the user. What happens when you write `true` or `false` in a YAML file when you meant `"true"` or `"false"`? Does something convert the boolean back to a string, or will the user get a type error? TBH I think I'd prefer almost any other way of solving this.

e.g. could we maybe only set the firmware settings if a Spec.Firmware != nil, but keep the settings themselves as booleans?

Reply:
If we're creating an enum, ideally we would come up with more descriptive values than "true" and "false". In this case, the field could be `Virtualization` and the values could be `enabled` and `disabled`.

Reply:
When I replace the "true" in the api with true, and then apply the following bmh:
```yaml
apiVersion: metal3.io/v1alpha1
kind: BareMetalHost
metadata:
  name: example-baremetalhost
spec:
  online: true
  bmc:
    address: irmc://192.168.30.29
    credentialsName: example-baremetalhost-secret
  bootMode: legacy
  bootMACAddress: 90:1b:0e:e6:bc:c1
  firmware:
    virtualizationEnabled: true
```

I got an error: ***The BareMetalHost "example-baremetalhost" is invalid: spec.firmware.virtualizationEnabled: Invalid value: "boolean": spec.firmware.virtualizationEnabled in body must be of type string: "boolean"***.

Then I replace ***virtualizationEnabled: true*** with ***virtualizationEnabled: "true"***, I got another error: ***The BareMetalHost "example-baremetalhost" is invalid: spec.firmware.virtualizationEnabled: Unsupported value: "true": supported values: "true", "false"***.

In my test
```go
	// +kubebuilder:validation:Enum="true";"false"
	VirtualizationEnabled string `json:"virtualizationEnabled,omitempty"`
```
and 
```go
	// +kubebuilder:validation:Enum=true;false
	VirtualizationEnabled bool `json:"virtualizationEnabled,omitempty"`
```
are the only two correct combinations. The corresponding user input is `virtualizationEnabled: "true"`(string) and `virtualizationEnabled: true`(bool).